### PR TITLE
Enhanced Secret Functionality

### PR DESCRIPTION
**Pull request by ejether**
This PR adds a new key to the course and new functionality for secret handling. The `secret` top level key is a list of maps that define how and hwere to get a secret value. The secrets are merged into the `os.environ` and that is used as before. Duplicate variables are not allowed. 

This pr adds two providers `AWSParameterStore` and `ShellExecutor`

### CODE REVIEW

**Comment by ejether** [comment]
I still need to add documentation and tests for the `ShellExecutor` but I wanted to get this out there sooner. 
I'd love feedback on the code structure, if there is any, before I start to go to town on the documentation. 
The way the secret provider frame is written feels like it might be wonky. with `SecretProvider.get_value()` not accepting arguments. 🤷 

All in all, this was really fun to write and it came together really nicely.

**Comment by ejether** [comment]
Closes: https://github.com/FairwindsOps/reckoner/issues/268

**Comment by ejether** [comment]
@pthomison you might be interested in this as well.

File: end_to_end_testing/course_files/13_test_lint_bad_secret.yml
Comment (ejether) [inline end_to_end_testing/course_files/13_test_lint_bad_secret.yml:10]:
```suggestion
- name: testname

```

File: end_to_end_testing/course_files/13_test_lint_good_secret.yml
Comment (ejether) [inline end_to_end_testing/course_files/13_test_lint_good_secret.yml:11]:
```suggestion
  allowed: test_other_allowed
  
```

File: end_to_end_testing/tests/13_schema.yaml
Comment (ejether) [inline end_to_end_testing/tests/13_schema.yaml:36]:
```suggestion
    - result.systemout ShouldContainString 'No schema validation errors found'

```

File: reckoner/tests/test_course.py
Comment (sudermanjr) [inline reckoner/tests/test_course.py:160]:
```suggestion
        # Doing this because the way secrets get injected into the environment
```

**Review by sudermanjr — CHANGES_REQUESTED** [review]
Seems good. Haven't tested myself, but the code looks good. I didn't notice, is there an e2e test that uses the shell executor to test that code path?

File: docs/README.md
Comment (ejether) [inline docs/README.md:196]:
```suggestion
```

File: reckoner/course.py
Comment (ejether) [reply reckoner/course.py:195]: Agreed, we will need some optimization if we add secrets at the chart and the global scope. 
This was the simplest and least disruptive way I could think of to accomplish the global scope.

**Comment by ejether** [comment]
There is not an e2e test for that. I'll add one.

**Review by sudermanjr — APPROVED** [review]
I am happy with this as it is. The structure seems sufficient for MVP, and it will solve most use-cases. Waiting for review from Luke before I merge.

**Review by lucasreed — APPROVED** [review]

**Pull request merged**
